Tools/dea_tools/classification.py

```diff
@@ -23,35 +23,36 @@
 import os
 import sys
 import joblib
-import datacube
-import rasterio
 import numpy as np
 import pandas as pd
 import xarray as xr
 import time
-from tqdm.auto import tqdm
-import dask.array as da
+import warnings
+from datetime import timedelta
 import geopandas as gpd
 from copy import deepcopy
+from tqdm.auto import tqdm
 import multiprocessing as mp
-import dask.distributed as dd
 import matplotlib.pyplot as plt
+
+from typing import Callable, Tuple, Any, Optional, List, Dict
+
+import dask.array as da
+from dask_ml.wrappers import ParallelPostFit
+import dask.distributed as dd
+from dask.diagnostics import ProgressBar
+
 from sklearn.cluster import KMeans
 from sklearn.utils import check_random_state
 from abc import ABCMeta, abstractmethod
-from datacube.utils import geometry
 from sklearn.base import ClusterMixin
-from dask.diagnostics import ProgressBar
-from rasterio.features import rasterize
-from dask_ml.wrappers import ParallelPostFit
 from sklearn.mixture import GaussianMixture
-from datacube.utils.geometry import assign_crs
 from sklearn.cluster import AgglomerativeClustering
 from sklearn.model_selection import KFold, ShuffleSplit
 from sklearn.model_selection import BaseCrossValidator
 
-import warnings
-
+from datacube.utils.geometry import assign_crs
```

Comment:
I'd love to switch all these funcs to use `odc.geo` instead of importing from `datacube.utils.geometry`, but I think that might be a job for a new PR

Reply:
This will be a separate PR.  I'm happy to do this so feel free to create a ticket and assign it to me.